web/src/portico/signup.ts

```diff
@@ -307,13 +307,12 @@ $(() => {
         $(e.target).hide();
     });
 
-    $("#how-realm-creator-found-zulip select").on("change", function () {
-        const elements: Record<string, string> = {
-            Other: "how-realm-creator-found-zulip-other",
-            Advertisement: "how-realm-creator-found-zulip-where-ad",
-            "At an organization that's using it":
-                "how-realm-creator-found-zulip-which-organization",
-        };
+    $<HTMLSelectElement>("#how-realm-creator-found-zulip select").on("change", function () {
+        const elements = new Map([
+            ["other", "how-realm-creator-found-zulip-other"],
+            ["ad", "how-realm-creator-found-zulip-where-ad"],
+            ["existing_user", "how-realm-creator-found-zulip-which-organization"],
+        ]);
```

Comment:
The keys in this map look like they're strings we expect to see in the app; I'm not sure it is safe to replace them.

Reply:
We previously looked this up by the `<option>` element’s text; we now use its value instead. I tested this.